src/platform/packages/shared/response-ops/rule_form/src/rule_definition/dashboard_service.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DashboardStart } from '@kbn/dashboard-plugin/public';
+import type { DashboardAttributes } from '@kbn/dashboard-plugin/server';
+
+export interface DashboardItem {
+  id: string;
+  attributes: Pick<DashboardAttributes, 'title'>;
+}
+
+export type DashboardService = ReturnType<typeof dashboardServiceProvider>;
+export type DashboardItems = Awaited<ReturnType<DashboardService['fetchDashboards']>>;
+
+export function dashboardServiceProvider(dashboardService: DashboardStart) {
+  return {
+    /**
+     * Fetch dashboards
+     */
+    async fetchDashboards(query?: string): Promise<DashboardItem[]> {
+      const findDashboardsService = await dashboardService.findDashboardsService();
+      const responses = await findDashboardsService.search({
+        search: query ? `${query}*` : '',
+        size: 1000,
```

Comment:
You can add the `spaces` option to specify an array of space names or `'*'` to search all available spaces. Only spaces for which the user has privileges will be searched.

```suggestion
        size: 1000,
        options: { spaces: ['*'] },
```

Additionally, you could specify `fields: ['title', 'description']` in the `options` to limit the search to only return specified `attributes` fields possibly making the response much smaller.

Reply:
@nickpeihl Thanks for your suggestion. I applied the change in this [commit](https://github.com/elastic/kibana/pull/209668/commits/5a235c2c4f4295a506c166bc777fb0840053f855)